### PR TITLE
writeBinaryCheckpoints method

### PR DESCRIPTION
**Pull request by tnitan**
- wrapped fileOutputStream in try/finally block and moved .close() calls  to avoid possible resource leaks

### CODE REVIEW

**Comment by schildbach** [comment]
Thanks for looking into this.

May I suggest to use the try-with-resources construct here? Then you could do without the manual closing at all. And while you're at it, you could do the same for the writeTextualCheckpoints() method.

**Comment by tnitan** [comment]
Sure, I've updated the pull-request. :)

File: tools/src/main/java/org/bitcoinj/tools/BuildCheckpoints.java [hunk 1]
```diff
@@ -172,42 +172,43 @@ public void notifyNewBestBlock(StoredBlock block) throws VerificationException {
...
-        dataOutputStream.writeInt(0);  // Number of signatures to read. Do this later.
-        digestOutputStream.on(true);
-        dataOutputStream.writeInt(checkpoints.size());
-        ByteBuffer buffer = ByteBuffer.allocate(StoredBlock.COMPACT_SERIALIZED_SIZE);
-        for (StoredBlock block : checkpoints.values()) {
-            block.serializeCompact(buffer);
-            dataOutputStream.write(buffer.array());
-            buffer.position(0);
+        try (FileOutputStream fileOutputStream = new FileOutputStream(file, false)) {
+            MessageDigest digest = Sha256Hash.newDigest();
```
Comment (schildbach) [inline tools/src/main/java/org/bitcoinj/tools/BuildCheckpoints.java]: I think if you move this line up above the try block, the two try blocks could be merged into one. Or am I overlooking something?

Comment (tnitan) [reply tools/src/main/java/org/bitcoinj/tools/BuildCheckpoints.java]: try-with-resources blocks can only use final variables that imlpement AutoCloseable interface, and the lines in second try are using: "MessageDigest digest = Sha256Hash.newDigest();" which doesn't meet the requirements.

Comment (tnitan) [reply tools/src/main/java/org/bitcoinj/tools/BuildCheckpoints.java]: You're right, I've moved them in a single try. Don't know why I didn't merge them in the first place. Thanks!

**Comment by schildbach** [comment]
Perfect, thanks! I just merged this.

**Pull request closed without merging**